python/semantic_kernel/connectors/memory/azure_cosmosdb/azure_cosmos_db_memory_store.py

```diff
@@ -0,0 +1,470 @@
+
+# Load environment variables
+(
+    cosmos_api,
+    cosmos_connstr,
+    cosmos_database_name,
+    cosmos_container_name,
+) = azure_cosmos_db_settings_from_dot_env()
+
+# OpenAI Ada Embeddings Dimension
+VECTOR_DIMENSION = 1536
+
+
+# Abstract class similar to the original data store that allows API level abstraction
+class AzureCosmosDBStoreApi(ABC):
+    @abstractmethod
+    async def ensure(self, num_lists, similarity, vector_dimensions):
+        raise NotImplementedError
+
+    @abstractmethod
+    async def create_collection_core(self, collection_name: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_collections_core(self) -> List[str]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def delete_collection_core(self, collection_name: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def does_collection_exist_core(self, collection_name: str) -> bool:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def upsert_core(self, collection_name: str, record: MemoryRecord) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def upsert_batch_core(
+        self, collection_name: str, records: List[MemoryRecord]
+    ) -> List[str]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_core(
+        self, collection_name: str, key: str, with_embedding: bool
+    ) -> MemoryRecord:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_batch_core(
+        self, collection_name: str, keys: List[str], with_embeddings: bool
+    ) -> List[MemoryRecord]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def remove_core(self, collection_name: str, key: str) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def remove_batch_core(self, collection_name: str, keys: List[str]) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_nearest_matches_core(
+        self,
+        collection_name: str,
+        embedding: ndarray,
+        limit: int,
+        min_relevance_score: float,
+        with_embeddings: bool,
+    ) -> List[Tuple[MemoryRecord, float]]:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_nearest_match_core(
+        self,
+        collection_name: str,
+        embedding: ndarray,
+        min_relevance_score: float,
+        with_embedding: bool,
+    ) -> Tuple[MemoryRecord, float]:
+        raise NotImplementedError
+
+
+class MongoStoreApi(AzureCosmosDBStoreApi):
+    def __init__(self, mongoClient: MongoClient):
+        self.collection = None
+        self.mongoClient = mongoClient
+
+    async def ensure(self, num_lists, similarity, vector_dimensions):
+        assert self.mongoClient.is_mongos
+        self.collection = self.mongoClient[cosmos_database_name][cosmos_container_name]
```

Comment:
It looks like this implementation works only with one container that is specified in configuration file. In `AzureCosmosDBMemoryStore` methods, parameter `collection_name` is not used, but I believe it should be. 

Let's imagine there is an application with entities like `Users`, `Documents`, `Devices` etc. 
In current implementation, I am forced to store all of my entities in one container, that I specify in configuration file. Even taking into account that we map all entities to `MemoryRecord`, we should provide a way for users to group these entities logically. If users want, they can still keep all entities in one container, but at least we will provide a way for more flexibility.

Ideally, I should be able to store my entities in different containers and have the ability to specify from my application the container I want to use. 
This will allow me to manage my database in flexible way, and each separate container could have its own set of stored procedures, triggers, user-defined functions etc. Other memory connectors in this repository are implemented in similar way.

Reply:
I have updated implementation according to the above discussion.